src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -1130,7 +1130,9 @@ public static bool IsLeapYear(int year)
             {
                 ThrowHelper.ThrowArgumentOutOfRange_Year();
             }
-            return (year & 3) == 0 && ((year & 15) == 0 || (year % 25) != 0);
+            return (year & 3) == 0 && ((year & 15) == 0 ||
+                   // TODO: optimize "(uint)year % 25 != 0" in JIT to produce:
```

Comment:
Yes, we should fix the JIT instead of sprinkling manually generated magic div and mod everywhere. I had the same comment in https://github.com/dotnet/runtime/pull/45479#discussion_r534386183

Reply:
ah didn't notice that pr 🙂